packages/jsii-pacmak/lib/targets/kotlin.ts

```diff
@@ -0,0 +1,66 @@
+import spec = require('jsii-spec');
+import { PackageInfo, Target } from '../target';
+import { shell } from '../util';
+import { KotlinGenerator } from './kotlin/kotlingenerator';
+import { getJvmPackageInfos, findJavaRuntimeLocalRepository } from './jvm/mavenutils';
+
+export class Kotlin extends Target {
+  public static toPackageInfos(assm: spec.Assembly): { [language: string]: PackageInfo } {
+    const kotlinTarget = assm.targets && assm.targets.kotlin;
+    const groupId = kotlinTarget && kotlinTarget.maven && kotlinTarget.maven.groupId;
+    const artifactId = kotlinTarget && kotlinTarget.maven && kotlinTarget.maven.artifactId;
+    const version = assm.version;
+    return getJvmPackageInfos('kotlin', groupId, artifactId, version);
+  }
+
+  public static toNativeReference(type: spec.Type, options: any) {
+    const [, ...name] = type.fqn.split('.');
+    return { java: `import ${[options.package, ...name].join('.')}` };
+  }
+
+  protected readonly generator = new KotlinGenerator();
+
+  public async build(sourceDir: string, outDir: string): Promise<void> {
+    const url = `file://${outDir}`;
+    const gradleArguments = new Array<string>();
+    for (const arg of Object.keys(this.arguments)) {
+      if (!arg.startsWith('gradle-')) { continue; }
+      gradleArguments.push(`--${arg.slice(4)}`);
+      gradleArguments.push(this.arguments[arg].toString());
+    }
+
+    const customRepoArgs = await this.getCustomRepoArgs(outDir);
+
+    await shell(
+      'gradle',
+      ['publish', `-PdeployRepo=${url}`, ...customRepoArgs, ...gradleArguments],
+      { cwd: sourceDir }
+    );
+  }
+
+  private async getCustomRepoArgs(outDir: string): Promise<string[]> {
+    const customRepos = await this.getCustomRepos(outDir);
+
+    const args = [];
+    let index = 1;
+    for (const repo of customRepos) {
+      args.push(`-PcustomRepo${index}=file://${repo}`);
+      index += 1;
+    }
+
+    return Promise.resolve(args);
```

Comment:
The method is `async` so you can simply return `args` here. Right now this gets wrapped twice (once explicitly, once implicitly due to  `async`).